src/main/java/jcifs/smb/SmbTreeConnectionTrace.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * © 2019 AgNO3 Gmbh & Co. KG
+ * 
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * 
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+package jcifs.smb;
+
+
+import jcifs.CIFSContext;
+
+
+/**
+ * @author mbechler
+ *
+ */
+class SmbTreeConnectionTrace extends SmbTreeConnection {
+
+    SmbTreeConnectionTrace ( CIFSContext ctx ) {
+        super(ctx);
+    }
+
+
+    SmbTreeConnectionTrace ( SmbTreeConnection conn ) {
+        super(conn);
+    }
+
+
+    @Override
+    protected void finalize () throws Throwable {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: This class extends the *SmbTreeConnection* superclass. If you do not invoke the finalizer on the superclass, its resources may not be disposed properly. Even when the superclass does not override the finalize() method, it is a best practice to invoke super.finalize() to prevent bugs in case the class hierarchy is changed in the future.

**Fix**: Invoke super.finalize() at the end of this method explicitly.